test/DotNetty.Transport.Tests.Performance/Sockets/TcpSocketChannelPerfSpec.cs

```diff
@@ -0,0 +1,122 @@
+    using DotNetty.Codecs;
+    using DotNetty.Transport.Bootstrapping;
+    using DotNetty.Transport.Channels;
+    using DotNetty.Transport.Channels.Sockets;
+    using DotNetty.Transport.Tests.Performance.Utilities;
+    using NBench;
+
+    public class TcpChannelPerfSpecs
+    {
+        const string InboundThroughputCounterName = "inbound ops";
+
+        const string OutboundThroughputCounterName = "outbound ops";
+
+        public const int WriteCount = 100000;
+
+        static readonly IPEndPoint TEST_ADDRESS = new IPEndPoint(IPAddress.IPv6Loopback, 0);
+        protected readonly ManualResetEventSlim ResetEvent = new ManualResetEventSlim(false);
+        IChannel _clientChannel;
+        Counter _inboundThroughputCounter;
+        Counter _outboundThroughputCounter;
+
+        IChannel _serverChannel;
+        IReadFinishedSignal _signal;
+
+        protected Bootstrap ClientBootstrap;
+
+        protected IEventLoopGroup ClientGroup;
+
+        byte[] message;
+        protected ServerBootstrap ServerBoostrap;
+        protected IEventLoopGroup ServerGroup;
+        protected IEventLoopGroup WorkerGroup;
+
+        static TcpChannelPerfSpecs()
+        {
+            // Disable the logging factory
+            //LoggingFactory.DefaultFactory = new NoOpLoggerFactory();
+        }
+
+        protected virtual IChannelHandler GetEncoder()
+        {
+            return new LengthFieldPrepender(4, false);
+        }
+
+        protected virtual IChannelHandler GetDecoder()
+        {
+            return new LengthFieldBasedFrameDecoder(20, 0, 4, 0, 4);
+        }
+
+        [PerfSetup]
+        public void SetUp(BenchmarkContext context)
+        {
+            this.ClientGroup = new MultithreadEventLoopGroup(1);
+            this.ServerGroup = new MultithreadEventLoopGroup(1);
+            this.WorkerGroup = new MultithreadEventLoopGroup();
+
+            Encoding iso = Encoding.GetEncoding("ISO-8859-1");
+            this.message = iso.GetBytes("ABC");
+
+            this._inboundThroughputCounter = context.GetCounter(InboundThroughputCounterName);
+            this._outboundThroughputCounter = context.GetCounter(OutboundThroughputCounterName);
+            var counterHandler = new CounterHandlerInbound(this._inboundThroughputCounter);
+            this._signal = new ManualResetEventSlimReadFinishedSignal(this.ResetEvent);
+
+            ServerBootstrap sb = new ServerBootstrap().Group(this.ServerGroup, this.WorkerGroup).Channel<TcpServerSocketChannel>()
+                .ChildHandler(new ActionChannelInitializer<TcpSocketChannel>(channel => { channel.Pipeline.AddLast(this.GetEncoder()).AddLast(this.GetDecoder()).AddLast(counterHandler).AddLast(new CounterHandlerOutbound(this._outboundThroughputCounter)).AddLast(new ReadFinishedHandler(this._signal, WriteCount)); }));
+
+            Bootstrap cb = new Bootstrap().Group(this.ClientGroup).Channel<TcpSocketChannel>().Handler(new ActionChannelInitializer<TcpSocketChannel>(
```

Comment:
does it make sense to use pooled allocator here as well? I.e. if we're not necessarily interested in performance of a single client here, it would speed things up. On the other hand it isn't necessarily a client scenario to use pooled allocator.